catalog/ui/search-ui/search-endpoint/src/test/java/org/codice/ddf/ui/searchui/query/controller/SearchControllerTest.java

```diff
@@ -390,6 +395,64 @@ public void testExecuteQueryCacheDisabled() throws Exception {
                 .size(), is(0));
     }
 
+    @Test
+    public void testMaxQueryPaging() throws Exception {
+        final long TOTAL_CATALOG_HITS = 1100L;
+        final int EXPECTED_MAX_RESULT = 1000;
+        final int SECOND_PAGE_RESULT = 500;
+        final int FIRST_PAGE_RESULT = 600;
+        final int EMPTY_PAGE = 0;
+        ArgumentCaptor<QueryResponse> queryResponseArgumentCaptor = ArgumentCaptor.forClass(
+                QueryResponse.class);
+        Search search = mock(Search.class);
+        when(mockQuery.getPageSize()).thenReturn(EXPECTED_MAX_RESULT);
+
+        setupModifiedFramework(TOTAL_CATALOG_HITS,
+                FIRST_PAGE_RESULT,
+                SECOND_PAGE_RESULT,
+                EMPTY_PAGE);
+        SourceQueryRunnable sourceQueryRunnable = setupSourceQueryRunnable(search);
+        sourceQueryRunnable.run();
+
+        verify(search).update(any(), queryResponseArgumentCaptor.capture());
+        assertThat("The framework should have only return " + TOTAL_CATALOG_HITS + " hits",
+                queryResponseArgumentCaptor.getValue()
+                        .getHits(),
+                is(TOTAL_CATALOG_HITS));
+        assertThat("The max query results was set to " + EXPECTED_MAX_RESULT,
+                queryResponseArgumentCaptor.getValue()
+                        .getResults()
+                        .size(),
+                is(EXPECTED_MAX_RESULT));
+    }
+
+    private SourceQueryRunnable setupSourceQueryRunnable(Search search) {
+        Subject subject = mock(Subject.class);
+        BayeuxServer bayeuxServer = mock(BayeuxServer.class);
+        ServerChannel channel = mock(ServerChannel.class);
+        when(bayeuxServer.getChannel(any(String.class))).thenReturn(channel);
+        searchController.setBayeuxServer(bayeuxServer);
+        when(mockSearchRequest.getQuery()).thenReturn(mockQuery);
+
+        return new SourceQueryRunnable(searchController,
+                "sampleSourceId",
+                mockSearchRequest,
+                subject,
+                new HashMap<>(),
+                search,
+                mockServerSession,
+                mockFuture,
+                mockFuture);
+    }
+
+    private void setupModifiedFramework(long totalHits, int... catalogResults) {
```

Comment:
👍  Great abstraction. Goes a long way to help with readability throughout test class.